resources/form-schemas/deployment-ui-schema.json

```diff
@@ -0,0 +1,158 @@
+{
+  "ui:title": "Service Properties",
+  "spec": {
+    "ui:title": "Object specs",
+    "selector": {
+      "ui:title": "Selector",
+      "matchLabels": {
+        "ui:title": "Match Labels"
+      }
+    },
+    "replicas": {
+      "ui:title": "Replicas"
+    },
+    "strategy": {
+      "ui:title": "Strategy",
+      "type": {
+        "ui:title": "Type"
+      }
+    },
+    "template": {
+      "ui:title": "Template",
+      "metadata": {
+        "ui:title": "Metadata",
+        "labels": {
+          "ui:title": "Labels"
+        }
+      },
+      "spec": {
+        "ui:title": "Spec",
+        "containers": {
+          "ui:title": "Containers",
+          "items": {
+            "name": {
+              "ui:title": "Name"
+            },
+            "image": {
+              "ui:title": "Image"
+            },
+            "ports": {
+              "ui:title": "Ports",
+              "items": {
+                "name": {
+                  "ui:title": "Name"
+                },
+                "containerPort": {
+                  "ui:title": "Container Port"
+                }
+              }
+            },
+            "resources": {
+              "limits": {
+                "ui:title": "Limits",
+                "memory": {
+                  "ui:title": "Memory"
+                },
+                "cpu": {
+                  "ui:title": "CPU"
+                }
+              },
+              "requests": {
+                "ui:title": "Requests",
+                "memory": {
+                  "ui:title": "Memory"
+                },
+                "cpu": {
+                  "ui:title": "CPU"
+                }
+              }
+            },
+            "env": {
+              "ui:title": "Environments",
+              "items": {
+                "name": {
+                  "ui:title": "Name"
+                },
+                "value": {
+                  "ui:title": "Value"
+                },
+                "valueFrom": {
+                  "ui:title": "Value From",
+                  "secretKeyRef": {
+                    "ui:title": "Secret Key Ref",
+                    "name": {
+                      "ui:title": "Name"
```

Comment:
I think this could be a widget for selecting existing from a list of existing secret names